crates/ol/block-assembly/src/context.rs

```diff
@@ -0,0 +1,63 @@
+//! Block assembly context trait.
+
+use strata_asm_common::AsmLogEntry;
+use strata_identifiers::OLTxId;
+use strata_ol_chain_types_new::{OLBlockId, OLTransaction};
+use strata_ol_state_types::OLState;
+use strata_primitives::l1::L1BlockCommitment;
+
+use crate::error::BlockAssemblyError;
+
+/// Context interface for block assembly.
+pub trait BlockAssemblyContext {
+    /// Gets a block's [`OLState`].
+    ///
+    /// Returns `Ok(None)` if block doesn't exist, `Err` with [`BlockAssemblyError`] if error
+    /// occurs.
+    fn get_ol_state(&self, block_id: OLBlockId) -> Result<Option<&OLState>, BlockAssemblyError>;
```

Comment:
It should be fine to return owned `OLState` here since it will be most likely be fetched from db and won't exist inside the trait implementor.